### PR TITLE
added DCOS attributes to agent pools

### PR DESCRIPTION
**Pull request by xtophs**
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/138)

### CODE REVIEW

File: pkg/api/convertertoapi.go [hunk 1]
```diff
@@ -174,6 +174,10 @@ func convertV20160330AgentPoolProfile(v20160330 *v20160330.AgentPoolProfile, api
 	api.FQDN = v20160330.FQDN
 	api.OSType = OSType(v20160330.OSType)
 	api.Subnet = v20160330.GetSubnet()
+	api.Attributes = map[string]string{}
```
Comment (anhowe) [inline pkg/api/convertertoapi.go]: only modify vlabs and not the other API versions

File: parts/dcoscustomdata187.t [hunk 2]
```diff
@@ -337,3 +337,5 @@ write_files:
   path: "/opt/azure/containers/provision.sh"
   permissions: "0744"
   owner: "root"
+- path: /var/lib/dcos/mesos-slave-common
```
Comment (anhowe) [inline parts/dcoscustomdata187.t]: are you also going to add this to dcoscustomdata173.t or validation to block 1.7.3's use of attributes?

File: examples/dcosattributes/dcos1.8.4.attributes.json [hunk 3]
```diff
@@ -0,0 +1,45 @@
+{
+  "apiVersion": "vlabs",
+  "properties": {
+    "orchestratorProfile": {
+      "orchestratorType": "DCOS184"
```
Comment (anhowe) [inline examples/dcosattributes/dcos1.8.4.attributes.json]: Two things in examples:
1. change this to "DCOS" only
2. add a README.md to this folder that describes the attributes for DCOS.  Only has to be a few sentences (what it does, and how to specify a container within DCOS).  An example README.md is under vnet/ example folder.

File: examples/dcosattributes/dcos1.8.4.attributes.json [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      "count": 1,
+      "dnsPrefix": "xtophmaster",
+      "vmSize": "Standard_D2_v2"
+    },
+    "agentPoolProfiles": [
+      {
+        "name": "xtentprivate",
+        "count": 3,
+        "vmSize": "Standard_D2_v2",
+        "attributes" : {
```
Comment (anhowe) [inline examples/dcosattributes/dcos1.8.4.attributes.json]: additionally, add validation to block adding this for other orchestrator types until we can plumb that through.

File: scripts/devenv.ps1 [hunk 5]
```diff
@@ -1,6 +1,6 @@
 $pwd = (Get-Location).Path
 
-docker build --pull -t acs-engine .
+#docker build --pull -t acs-engine .
```
Comment (JackQuincy) [inline scripts/devenv.ps1]: I'm guessing this was for local debugging. Please uncomment this line

**Comment by brendandburns** [comment]
Can one of the admins verify this patch?

**Comment by JackQuincy** [comment]
This PR was restarted and that PR has been merged. https://github.com/Azure/acs-engine/pull/362

**Pull request closed without merging**